cirq-core/cirq/work/sampler.py

```diff
@@ -293,11 +294,18 @@ async def run_batch_async(
 
         See docs for `cirq.Sampler.run_batch`.
         """
+        print("finally")
         params_list, repetitions = self._normalize_batch_args(programs, params_list, repetitions)
-        return [
-            await self.run_sweep_async(circuit, params=params, repetitions=repetitions)
-            for circuit, params, repetitions in zip(programs, params_list, repetitions)
-        ]
```

Comment:
You probably want to use a variant of `duet.pmap` ("parallel map") to make multiple calls in parallel. In this case, the `zip` produces tuples of args and we want to call the underlying function like `self.run_sweep_async(*args)`, so we need a "starmap", something like:
```python
return await duet.pstarmap_async(self.run_sweep_async, zip(programs, params_list, repetitions))
```

Reply:
For reference, here's the definition of `duet.pstarmap_async`: https://github.com/google/duet/blob/main/duet/api.py#L171

Reply:
Thank you for the code pointer!